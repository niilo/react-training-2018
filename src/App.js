import React, {Component} from 'react';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import Home from './components/Home';
import Nav from './components/Nav';
import Battle from './components/Battle';
import Results from './components/Results';
import Popular from './components/Popular';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div className="container">
          <Nav />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/battle" component={Battle} />
            <Route path="/battle/results" component={Results} />
            <Route path="/popular" component={Popular} />
            <Route
              render={() => {
                return <p>Not found</p>;
              }}
            />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
