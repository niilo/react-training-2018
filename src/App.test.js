import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders correctly', () => {
  const mockResponse = (status, statusText, response) => {
    return new window.Response(response, {
      status,
      statusText,
      headers: {
        'Content-type': 'application/json',
      },
    });
  };

  window.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve(
      mockResponse(
        200,
        null,
        `{
        "repos": [
          {
            "name": "react",
            "html_url": "https://github.com/facebook/react",
            "stargazers_count": 73600,
            "owner": {
              "login": "facebook",
              "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4"
            }
          },
          {
            "name": "react2",
            "html_url": "https://github.com/facebook/react",
            "stargazers_count": 73600,
            "owner": {
              "login": "facebook",
              "avatar_url": "https://avatars3.githubusercontent.com/u/69631?v=4"
            }
          }
        ]
      }`,
      ),
    ),
  );

  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
