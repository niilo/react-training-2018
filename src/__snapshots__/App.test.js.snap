// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <ul
    className="nav"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/battle"
        onClick={[Function]}
        style={undefined}
      >
        Battle
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        className={undefined}
        href="/popular"
        onClick={[Function]}
        style={undefined}
      >
        Popular
      </a>
    </li>
  </ul>
  <div
    className="home-container"
  >
    <h1>
      Gihub Battle: Battle your friends:
    </h1>
    <a
      className="button"
      href="/battle"
      onClick={[Function]}
    >
      Battle
    </a>
  </div>
</div>
`;
